recipes/mimalloc/all/test_package/main.cpp

```diff
@@ -0,0 +1,16 @@
+#include <memory>
+#include <mimalloc.h>
+
+struct Test
+{ };
+
+int main()
+{
+  auto t0 = std::make_shared< Test >();
+
+  {
+    auto t1 = std::make_unique< Test >();
+  }
+
+  return 0;
+}
```

Comment:
mimalloc is a c library.
I would make the test c only (==> test_package.c)
https://github.com/microsoft/mimalloc/blob/master/include/mimalloc.h#L145-L148

```suggestion
#include "mimalloc.h"

#include <stdlib.h>
#include <stdio.h>

int main() {
    void *data = malloc(1024);
    free(data);

    std::cout << "mimalloc version " << mi_version() << "\n";
    mi_stats_print(nullptr);
    return 0;
}
```

Reply:
Ok, now i'm stumbling on the `cpp_info.libs` issue, which now is failing when assigned to `tools.collect_libs(self)`. I'm listing the libs explicitly then.

Reply:
BTW, thank you for the detailed review :)

Reply:
No problem.
There is a thin line between nagging and reviewing :smile: :angel:

Reply:
> Ok, now i'm stumbling on the `cpp_info.libs` issue, which now is failing when assigned to `tools.collect_libs(self)`. I'm listing the libs explicitly then.

The next problem will probably be about shared builds on Windows.
I think we also need to set `self.cpp_info.bindirs` to the same value as `self.cpp_info.libdirs`.
But let us allow the ci to finish first.